### ISSUE TITLE
Hyprpaper segfaults while setting wallpaper

### ISSUE THREAD

**Issue by FjorgeTheHuman** [hyprwm/hyprpaper#51]
### Software Versions
- Arch Linux (Kernel 6.2.2-zen1-1-zen)
- Hyprland: https://github.com/hyprwm/Hyprland/commit/4f647a8e8be78d3817f126e41940bef49e81232e (with [xwayland: support HiDPI scale](https://aur.archlinux.org/cgit/aur.git/tree/0001-xwayland-support-HiDPI-scale.patch?h=hyprland-hidpi-xprop-git) and [Fix configure_notify event](https://aur.archlinux.org/cgit/aur.git/tree/0002-Fix-configure_notify-event.patch?h=hyprland-hidpi-xprop-git) patches)
- Hyprpaper: 83867464c523c3ac2f7ce53d2c4b8d4f9b7551cd
---
### Issue
Whenever I run `hyprpaper`, it crashes with the following:
```
[LOG] Welcome to hyprpaper!
built from commit 83867464c523c3ac2f7ce53d2c4b8d4f9b7551cd (?)
[LOG] Cleaned old hyprpaper preloads (2), removing 27.1MB
[LOG] Preloaded target /home/fjorge/.config/hypr/wallpapers/forest-with-river.jpg in 284.23ms -> Pixel size: [5312, 2988]
[LOG] hyprpaper socket started at /tmp/hypr/4f647a8e8be78d3817f126e41940bef49e81232e_1678124441/.hyprpaper.sock (fd: 4)
[LOG] configure for eDP-1
[LOG] handlePreferredScale: 1.40 for 564a9f2a15f0
[LOG] Buffer created for target /home/fjorge/.config/hypr/wallpapers/forest-with-river.jpg, Shared Memory usage: 13.6MB
[LOG] Total SM usage for all buffers: 13.6MB
[LOG] Pool buffer missing for available target??
[LOG] Buffer created for target /home/fjorge/.config/hypr/wallpapers/forest-with-river.jpg, Shared Memory usage: 13.6MB
[LOG] Total SM usage for all buffers: 27.1MB
zsh: segmentation fault (core dumped)  hyprpaper
```
My hyprpaper configuration file consists of:
```ini
preload = ~/.config/hypr/wallpapers/forest-with-river.jpg

wallpaper = ,~/.config/hypr/wallpapers/forest-with-river.jpg
```
Running `coredumpctl gdb hyprpaper` and then `bt` within the gdb prompt gives me:
```gdb
#0  0x0000561cbf28b229 in CHyprpaper::renderWallpaperForMonitor(SMonitor*) ()
#1  0x0000561cbf28ca03 in CHyprpaper::tick(bool) ()
#2  0x00007f1891cc94f6 in ffi_call_unix64 () at ../src/x86/unix64.S:104
#3  0x00007f1891cc5f5e in ffi_call_int
    (cif=cif@entry=0x7ffc0a886ce0, fn=<optimized out>, rvalue=<optimized out>, avalue=<optimized out>, closure=closure@entry=0x0) at ../src/x86/ffi64.c:673
#4  0x00007f1891cc8b73 in ffi_call (cif=cif@entry=0x7ffc0a886ce0, fn=<optimized out>, rvalue=rvalue@entry=0x0, avalue=avalue@entry=0x7ffc0a886db0)
    at ../src/x86/ffi64.c:710
#5  0x00007f18922a9645 in wl_closure_invoke (closure=closure@entry=0x561cc0127d50, target=<optimized out>, 
    target@entry=0x561cc01278c0, opcode=opcode@entry=0, data=<optimized out>, flags=1) at ../wayland-1.21.0/src/connection.c:1025
#6  0x00007f18922a9e03 in dispatch_event (display=display@entry=0x561cc0121190, queue=0x561cc0121260) at ../wayland-1.21.0/src/wayland-client.c:1595
#7  0x00007f18922a9ffc in dispatch_queue (queue=0x561cc0121260, display=0x561cc0121190) at ../wayland-1.21.0/src/wayland-client.c:1741
#8  wl_display_dispatch_queue_pending (display=0x561cc0121190, queue=0x561cc0121260) at ../wayland-1.21.0/src/wayland-client.c:1983
#9  0x0000561cbf28cd06 in CHyprpaper::init() ()
#10 0x0000561cbf286db9 in main ()
```
---
### Additional Information
https://github.com/swaywm/swaybg works perfectly fine.

**Comment by Fxzzi** [comment]
This seems to only happen with decimal scaling values like you are using (1.4). I use 1.333 so that my 1440p monitor is the same scale as my 1080p one but alas I experience the same issue.

**Comment by vaxerski** [comment]
Thought that was fixed. Odd.

**Comment by FjorgeTheHuman** [comment]
> This seems to only happen with decimal scaling values like you are using (1.4). I use 1.333 so that my 1440p monitor is the same scale as my 1080p one but alas I experience the same issue.

I tried it and you seem to be right, except for some reason it works with a scaling value of 1.5 as well. It doesn't seem to work with other *.5 values though.

**Comment by LrsChrSch** [comment]
Have the same issue on a 4k monitor with a scaling of 1.6
1.5 works fine, my other FullHD monitor with a scaling of 1 also doesn't have any problems.

So not adding anything new here, this is just a little push at best :]

**Comment by emsbro100** [comment]
Can confirm I'm having the same issue on a 1440p monitor with scaling of 1.33.

**Comment by alerque** [comment]
Also can confirm, displays scaled at 1 work all the time, and some other scalings work some of the time. Some always cause segfaults.

**Comment by Fxzzi** [comment]
I've since switched to swaybg and everything is working gr8

**Comment by vaxerski** [comment]
fixed in 756fccfc53af80254f21db119d833b1caf9b7221
